src/clib/pioc.c

```diff
@@ -1698,14 +1698,16 @@ int PIOc_iotype_available(int iotype)
  * @ingroup PIO_init
  * @author Ed Hartnett
  */
-int PIOc_init_async(MPI_Comm world, int num_io_procs, int *io_proc_list,
-                    int component_count, int *num_procs_per_comp, int **proc_list,
+int PIOc_init_async(MPI_Comm world, int num_io_procs, const int *io_proc_list,
+                    int component_count, const int *num_procs_per_comp, const int **proc_list,
                     MPI_Comm *user_io_comm, MPI_Comm *user_comp_comm, int rearranger,
                     int *iosysidp)
 {
     int my_rank;          /* Rank of this task. */
-    int **my_proc_list;   /* Array of arrays of procs for comp components. */
-    int *my_io_proc_list; /* List of processors in IO component. */
+    const int **my_proc_list = NULL;   /* Array of arrays of procs for comp components. */
+    int **dummy_proc_list = NULL;      /* A temp buffer in case the user does not provide the proc list. */
```

Comment:
Can you move the dummy_*_list inside the if block (its not really needed outside the block, right?)?

Reply:
@jayeshkrishna It is needed outside the block for free() call, which has compiler warnings on pointers to consts.
Example test code:
```
#include <stdlib.h>

int main()
{
  const int *p = NULL;
  free(p);

  return 0;
}
```
```
test.c: In function ‘main’:
test.c:6:8: warning: passing argument 1 of ‘free’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   free(p);
        ^
In file included from test.c:1:0:
/usr/include/stdlib.h:563:13: note: expected ‘void *’ but argument is of type ‘const int *’
 extern void free (void *__ptr) __THROW;
             ^~~~
```

Reply:
ok, thanks

Reply:
Can we rename dummy_proc_list to something else (proc_list_buf or something like that) ? IMO the malloced proc list is not really a dummy proc list since it is initialized and used in the function.

Reply:
OK, I have renamed dummy_proc_list/dummy_io_proc_list.